homeassistant/components/meteoclimatic/__init__.py

```diff
@@ -0,0 +1,103 @@
+"""Support for Meteoclimatic weather data."""
+import asyncio
+import datetime
+import logging
+
+from meteoclimatic import MeteoclimaticClient
+from meteoclimatic.exceptions import MeteoclimaticError
+import voluptuous as vol
+
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.typing import ConfigType, HomeAssistantType
+from homeassistant.util import Throttle
+
+from .const import CONF_STATION_CODE, DOMAIN, PLATFORMS
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = datetime.timedelta(minutes=5)
+
+
+STATION_CODE_SCHEMA = vol.Schema({vol.Required(CONF_STATION_CODE): cv.string})
+
+CONFIG_SCHEMA = vol.Schema(
+    {DOMAIN: vol.Schema(vol.All(cv.ensure_list, [STATION_CODE_SCHEMA]))},
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+async def async_setup(hass: HomeAssistantType, config: ConfigType) -> bool:
+    """Set up Meteoclimatic from legacy config file."""
+
+    conf = config.get(DOMAIN)
+    if conf is None:
+        return True
+
+    for station_code_conf in conf:
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN, context={"source": SOURCE_IMPORT}, data=station_code_conf.copy()
+            )
+        )
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistantType, entry: ConfigEntry) -> bool:
+    """Set up an Meteoclimatic account from a config entry."""
+    hass.data.setdefault(DOMAIN, {})
+
+    # Weather
+    station_code = entry.data[CONF_STATION_CODE]
+    client = MeteoclimaticClient()
+    hass.data[DOMAIN][station_code] = MeteoclimaticUpdater(client, station_code)
+    await hass.async_add_executor_job(hass.data[DOMAIN][station_code].update)
+
+    for platform in PLATFORMS:
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(entry, platform)
+        )
+    _LOGGER.debug("meteoclimatic sensor platform loaded for %s", station_code)
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistantType, entry: ConfigEntry):
+    """Unload a config entry."""
+    unload_ok = all(
+        await asyncio.gather(
+            *[
+                hass.config_entries.async_forward_entry_unload(entry, platform)
+                for platform in PLATFORMS
+            ]
+        )
+    )
+    if unload_ok:
+        hass.data[DOMAIN].pop(entry.data[CONF_STATION_CODE])
+
+    return unload_ok
+
+
+class MeteoclimaticUpdater:
```

Comment:
Since you have a single API call common to all entities, you could consider using DataUpdateCoordinator as in https://developers.home-assistant.io/docs/integration_fetching_data#coordinated-single-api-poll-for-data-for-all-entities

Reply:
Thanks for suggestion. I've refactored to use DataUpdateCoordinator.